Manage Favorites from Favorites Overlay

afterxleep opened the pull request:
Task/Issue URL: https://app.asana.com/0/414709148257752/1203858036570302/f

### Description:
We allow editing and reordering favorites from the HomeViewController and the Bookmarks Manager but not from the Favorites Overlay.

This change allows Reordering, Editing, and Removing Favorites from the Favorites Overlay, and removes logic around `textFieldDidEndEditing(_:)` for the Omnibar to prevent the Favorites view from being dismissed while interacting with favorites.

#### Bonus:
Changes different strings in MainViewController to Constants

### Steps to test this PR:
1. Fire up the app (Make sure you have a few 'Favorites' added)
2. Navigate to any website (Important ⚠️)
3. Tap the Omnibar
4. Long-Press on one of your favorites
5. Observe that the 'Edit' / 'Delete' popover appears
6. **Test the functionality of both 'Edit' and 'Delete' from that view.**

7. Navigate to another website
8. Tap the omnibar
9. Reorder Your favorites via drag/drop
10. Tap 'Cancel.'
11. Tap the omnibar
12. **Observe that Favorites are ordered as expected**

13. Navigate to a few more websites
14. Tap the Ombibar
15. Tap the 'Back' button
16. **Observe the Omnibar is dismissed and the cancel button is no longer visible**
17. Repeat these steps 13-16 with all buttons in the Tab Bar.

### Demo
![demo](https://user-images.githubusercontent.com/1156669/226630082-973bafde-9d79-4cc3-8e4e-51f31860e1f3.gif)



---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**

afterxleep committed:
Allow Edit/Delete favorites from Overlay

afterxleep committed:
Rename ‘viewModel’ to ‘favoritesViewModel’

afterxleep committed:
Remove misleading comment

afterxleep committed:
Move strings to Constants

afterxleep committed:
Updated segue name

afterxleep committed:
Add delegate methods for Drag & Drop in Overlay

afterxleep committed:
Remove onDismiss

afterxleep committed:
Manually dismiss onmibar

Previously this behavior was depended on text not being edited anymore

afterxleep committed:
Refresh Favorites Overlay on ViewWillAppear

afterxleep committed:
Update favorites when external updates happen

afterxleep committed:
Make sure we submit a query when hitting return on the keyboard

afterxleep committed:
Lint fix

amddg44 commented on DuckDuckGo/FavoritesOverlay.swift:
```diff
@@ -40,11 +42,16 @@ class FavoritesOverlay: UIViewController {
     
     private var theme: Theme!
     
+    fileprivate var viewModelCancellable: AnyCancellable?
```
Minor nit - is there any reason why this can't be made `private` instead?

afterxleep replied:
Fixed, thanks.  Not really..  Just follow the pattern I saw in other bookmark mgmt places, but also not likely needed there.

amddg44 reviewed (approved):
LGTM! Just one small nit but otherwise everything is looking good and performs well 🙂

afterxleep committed:
Switch cancellable to be private

afterxleep committed:
Add constraints to collectionView

Manually adds constraints to the view looks the same as the FavoritesViewController

amddg44 commented on DuckDuckGo/FavoritesOverlay.swift:
```diff
@@ -79,9 +80,31 @@ class FavoritesOverlay: UIViewController {
+
+        // Size constraints
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        let widthConstraint = collectionView.widthAnchor.constraint(lessThanOrEqualToConstant: Constants.collectionViewMaxWidth)
```
@afterxleep Instead of using a hard coded value, perhaps it would be better to use the width of the view (maybe minus margins) for the width anchor constant to better support all sizes of devices?

afterxleep replied:
@amddg44  This was my first approach but I wanted the view to look the same as the HomeView controller favorites for consistency.   I could change both views to be the same way, but not sure that'd be part of the scope.

afterxleep committed:
Use full-width of the overlay on iPads

amddg44 reviewed (approved):
@afterxleep changes LGTM!

afterxleep committed:
Merge branch 'develop' into daniel/manage-favorites-overlay

afterxleep committed:
Merge branch 'develop' into daniel/manage-favorites-overlay
